### PR TITLE
Apache

### PR DESCRIPTION
**Pull request by halleyzhao**

### CODE REVIEW

File: v4l2/v4l2_codecbase.cpp [hunk 1]
```diff
@@ -391,7 +391,7 @@ int32_t V4l2CodecBase::ioctl(int command, void* arg)
                 ERROR("unknown request buffer type: %d", reqbufs->type);
                 break;
             }
-            ASSERT(port == INPUT || reqbufs->memory == m_memoryMode[port]);
+            // ASSERT(port == INPUT || reqbufs->memory == m_memoryMode[port]);
```
Comment (xuguangxin) [inline v4l2/v4l2_codecbase.cpp:394]: why remove this assert, do we need some comment?

**Comment by xuguangxin** [comment]
Most ok to me.Wait for comment about assert

File: v4l2/v4l2_codecbase.cpp (hunk 1 above)
Comment (halleyzhao) [inline v4l2/v4l2_codecbase.cpp:394]: VIDIOC_REQBUFS init memory type of input/output buffer.
1.  chromeos assumed input/output buffer memory type is MMAP, so we init m_memoryModep[port] to MMAP and ASSERT() here
2. next, we add USRPTR memory type for input buffer, it means 'port == INPUT' needn't assert memory type (but accept memmory type set from client).
3. now we add ANDROID_BUFFER_HANDLE memory type for output buffer, then 'port == OUTPUT' needn't ASSERT memory type either.

**Comment by xuguangxin** [comment]
ok, +1

File: interface/VideoCommonDefs.h [hunk 2]
```diff
@@ -94,7 +94,7 @@ typedef enum {
     VIDEO_DATA_MEMORY_TYPE_DMA_BUF,      // share buffer with camera device etc
     VIDEO_DATA_MEMORY_TYPE_SURFACE_ID,  // it can be used for surface sharing of transcoding, benefits suppressed rendering as well.
                                         //it is discouraged to use it for video rendering.
-    VIDEO_DATA_MEMORY_TYPE_ANDROID_NATIVE_BUFFER, // ANativeWindowBuffer for android
```
Comment (xuguangxin) [inline interface/VideoCommonDefs.h:97]: We need increase API version for this  in configure.ac

**Comment by xuguangxin** [comment]
Hi @halleyzhao 
Please change 
`m4_define([yami_api_micro_version], 2)`
to
`m4_define([yami_api_micro_version], 3)`
in configure.ac

**Comment by xuguangxin** [comment]
merged as dd36dee96a218fa04ccc1c94ea74565821e99019. thanks @halleyzhao

**Pull request closed without merging**